Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -472,6 +473,11 @@ void getSupportedTypes(HashSet<String, ASCIICaseInsensitiveHash>& types) const f
     return synchronizerTime;
 }
 
+bool MediaPlayerPrivateMediaSourceAVFObjC::currentMediaTimeMayProgress() const
+{
+    return MediaSourcePrivate::hasFutureTime(currentMediaTime(), durationMediaTime(), *buffered());
```

Comment:
How do we know buffered() is not null?
It is also a bit sad to allocate some memory in the heap to just free it right away.
optional<> or Markable would be better there.

Reply:
it's backed by a UniqueRef<PlatformTimeRanges>

Reply:
I will change the prototyping of this method , it's annoying me too